### DISCUSSION TITLE
The error : PhysX error: NpSceneQueries::raycast pose is not valid.

### DISCUSSION THREAD

**Discussion by jimmyfu0729** [isaac-sim/IsaacLab#89] [Q&A] [answered]
I have built some environments like pepper, drone in the orbit. Their actuators are linear drives, and they work very well. This time I started to build carter environments. However, when the scene is reset, some of the carters will have strange behaviours like spiral lifting or rotating. When this situation happens, I have the errors from the PhysX about NpSceneQueries. I have checked the documentation of the PhysX and found that this is a type of collisions query of PhysX. I'm not sure how to solve the problems like this. By the way, when the velocity of the wheels increase or the dt of the scene increase to 0.2s this problem will increase significantly. Is there any hints why this problem happens?

**Comment by jimmyfu0729** [comment]
Are there any suggestions in tuning the PhysX parameters? I have no very clear ides in these parameters like gpu_found_lost_aggregate_pairs_capacity, reading PhysX documentation is very time consuming.

**Reply by Mayankm96** [reply, marked as answer]
Those parameters shouldn't really affect the error that you get. They are mainly for GPU buffer allocation which usually don't need to be changed unless you change the scene complexity.

Are you using LiDAR sensors or some kind of raycast operations in your environments? Maybe check if enabling this flag helps:
> https://github.com/NVIDIA-Omniverse/Orbit/blob/main/source/extensions/omni.isaac.orbit_envs/omni/isaac/orbit_envs/isaac_env_cfg.py#L160-L173

**Reply by jimmyfu0729** [reply]
Thanks, it works. But another problem is sometimes the observation buffer of some environments becomes NaN and this problem just happens in the Carter environment, I don't know if this is a specific problem on this agent or a common problem. But I didn't get this problem in other environments. Here I have 128 envs.
 f"Expected parameter {param} "
ValueError: Expected parameter loc (Tensor of shape (128, 2)) of distribution Normal(loc: torch.Size([128, 2]), scale: torch.Size([128, 2])) to satisfy the constraint Real(), but found invalid values:
tensor([[    -0.2341,     -0.0961],
        [    -0.2775,     -0.4407],
        [    -0.4888,     -0.5703],
        [     0.2214,     -0.0445],
        [    -0.0836,     -0.0890],
        [     0.1530,     -0.1884],
        [    -0.7937,     -0.8329],
        [    -0.4032,     -0.2741],
        [    -0.2896,     -0.3062],
        [    -0.2964,     -0.0143],
        [     0.1556,     -0.1638],
        [     0.3404,     -0.2050],
        [    -0.0475,     -0.0552],
        [    -0.1839,     -0.0465],
        [    -0.4115,     -0.2677],
        [    -0.3915,     -0.4847],
        [     0.0108,     -0.1779],
        [        nan,         nan],
        [     0.2332,     -0.1370],
        [    -0.1672,     -0.2364],
        [    -0.2690,     -0.2665],
        [    -0.5790,     -0.7035],
        [    -0.3037,     -0.0460],
        [    -0.1232,     -0.0868],
        [     0.0525,     -0.1427],
        [     0.0540,     -0.1963],
        [     0.1353,     -0.1200],
        [     0.0172,     -0.2874],
        [     0.1686,     -0.1642],
        [     0.3621,     -0.0615],
        [     0.2054,     -0.1446],
        [    -0.0808,     -0.1052],
        [     0.3489,     -0.1133],
        [    -0.1980,     -0.1673],
        [     0.2648,     -0.1900],
        [     0.2750,     -0.2314],
        [     0.0646,     -0.2257],
        [    -0.4892,     -0.3962],
        [    -0.1691,     -0.2487],
        [    -0.0519,     -0.2917],
        [     0.1302,     -0.1779],
        [     0.0686,     -0.1560],
        [     0.2024,     -0.0927],
        [    -0.0527,     -0.1055],
        [     0.0021,     -0.2174],
        [     0.2082,     -0.0874],
        [    -0.4809,     -0.5379],
        [    -0.6135,     -0.5082],
        [     0.1549,     -0.0739],
        [    -0.3187,     -0.0877],
        [     0.1665,     -0.1355],
        [     0.2458,     -0.1757],
        [    -0.0950,     -0.0643],
        [     0.3563,     -0.1275],
        [    -0.0303,     -0.0983],
        [     0.1535,     -0.1502],
        [    -0.1743,     -0.1645],
        [     0.0633,     -0.1777],
        [    -0.1533,     -0.0127],
        [     0.0516,     -0.1302],
        [    -0.6215,     -0.7149],
        [    -0.0556,     -0.1646],
        [     0.0171,     -0.1363],
        [    -0.2187,     -0.5258],
        [    -0.0861,     -0.1541],
        [    -0.2676,     -0.1681],
        [     0.0414,     -0.1820],
        [     0.0808,     -0.1796],
        [     0.1264,     -0.1614],
        [     0.4344,     -0.1650],
        [     0.0968,     -0.2005],
        [    -0.0079,     -0.2095],
        [     0.0750,     -0.1156],
        [     0.1559,     -0.1231],
        [    -0.1284,     -0.0345],
        [     0.3831,     -0.0435],
        [     0.1877,     -0.2243],
        [    -0.0650,     -0.0794],
        [    -0.7090,     -0.7126],
        [     0.1441,     -0.1126],
        [    -0.0023,     -0.1109],
        [    -0.1733,     -0.1215],
        [     0.0115,     -0.1654],
        [     0.1322,     -0.1709],
        [     0.0764,     -0.1032],
        [     0.0484,     -0.2800],
        [     0.3717,     -0.1460],
        [    -0.0432,     -0.2184],
        [    -0.0006,     -0.1239],
        [    -0.0249,     -0.1806],
        [    -0.4643,     -0.3210],
        [    -0.0490,     -0.1362],
        [    -0.9219,     -0.8278],
        [     0.0667,     -0.1958],
        [    -0.2072,     -0.2076],
        [     0.1400,     -0.1909],
        [     0.1315,     -0.1363],
        [     0.2508,     -0.1629],
        [    -0.8755,     -0.6387],
        [    -0.0089,     -0.2627],
        [     0.3572,     -0.1277],
        [    -0.0494,     -0.0725],
        [    -0.0404,     -0.4379],
        [    -0.3632,     -0.4894],
        [     0.0516,     -0.1545],
        [     0.2629,     -0.0601],
        [     0.4681,     -0.1270],
        [     0.2158,     -0.1892],
        [     0.4122,     -0.1110],
        [    -0.0618,     -0.0878],
        [     0.0344,     -0.1255],
        [    -0.0398,     -0.1883],
        [     0.3361,     -0.1626],
        [     0.0403,     -0.1665],
        [    -0.1514,     -0.1203],
        [    -0.2385,     -0.3915],
        [    -0.5758,     -0.5616],
        [    -0.0440,     -0.1493],
        [     0.3152,     -0.1275],
        [    -0.6314,     -0.3033],
        [     0.1329,     -0.1929],
        [     0.4013,     -0.0901],
        [    -0.1945,     -0.0173],
        [    -0.0004,     -0.1794],
        [     0.3402,     -0.1809],
        [     0.2103,     -0.0439],
        [    -0.2492,     -0.0987],
        [     0.1243,     -0.1701]], device='cuda:0')

**Reply by Mayankm96** [reply]
I have seen this issue sometimes in some other environments. This typically happens when bodies are overlapping at resets or penetrating too much into each other. In those cases, the GPU physics tends to go "unstable" and NaN values occur.

My suggestions are:
* Check that your reset happens properly and there are no overlapping bodies
* Check that you get NaNs when running on CPU as well

The NaNs on GPU physics is an issue we have told the physics team and they are looking into fixing this for the coming release of Isaac Sim.
